src/components/timers/TimerSettings.tsx

```diff
@@ -0,0 +1,119 @@
+import { Input } from '../generic/Input';
+
+interface TimerSettingsProps {
+    type: string;
+    minValue?: number;
+    secValue?: number;
+    workMinValue?: number;
+    workSecValue?: number;
+    restMinValue?: number;
+    restSecValue?: number;
+    roundsValue?: number;
+    setMinValue?: (value: number) => void;
+    setSecValue?: (value: number) => void;
+    setWorkMinValue?: (value: number) => void;
+    setWorkSecValue?: (value: number) => void;
+    setRestMinValue?: (value: number) => void;
+    setRestSecValue?: (value: number) => void;
+    setRoundsValue?: (value: number) => void;
+}
+
+const TimerSettings = ({ 
+    type,
+    minValue,
+    secValue,
+    workMinValue,
+    workSecValue,
+    restMinValue,
+    restSecValue,
+    roundsValue,
+    setMinValue,
+    setSecValue,
+    setWorkMinValue,
+    setWorkSecValue,
+    setRestMinValue,
+    setRestSecValue,
+    setRoundsValue 
+}: TimerSettingsProps) => {
+
+    const xyTimer = type === 'XY';
+    const tabataTimer = type === 'Tabata';
+    
+    // Display timer settings inputs
+    return (
```

Comment:
This works for the current set of timers and there's nothing wrong with it. Now imagine you had add another type of timer, or 3 more. The settings here are a bit intertwined, what would be a bit cleaner would be something closer to a factory pattern, where each type of timer defines its own form with its settings. Here you would simply check the timer type and call get `<TabataSettings/>`, `<StopwatchSettings/>`, etc. Some would share inputs (Tabata and XY would both use Rounds) but they would be independent and adding new ones would be very simple.